From 4159f984bd351caf6f0e1adbd90feeb85ab252f4 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Wed, 16 Sep 2020 22:02:04 +0200
Subject: [PATCH] restore compatibility with networkmanager 1.22

drop this patch and rebuild gnome-control-center when updating nm
---
 panels/network/net-device-wifi.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git panels/network/net-device-wifi.c panels/network/net-device-wifi.c
index cb86e5b..d63c57c 100644
--- panels/network/net-device-wifi.c
+++ panels/network/net-device-wifi.c
@@ -1180,11 +1180,6 @@ nm_client_on_permission_change (NetDeviceWifi *self) {
         NMClientPermissionResult perm;
         NMDeviceWifiCapabilities caps;
 
-        if (nm_client_get_permissions_state (self->client) != NM_TERNARY_TRUE) {
-                /* permissions aren't ready yet */
-                return;
-        }
-
         /* only enable the button if the user can create a hotspot */
         perm = nm_client_get_permission_result (self->client, NM_CLIENT_PERMISSION_WIFI_SHARE_OPEN);
         caps = nm_device_wifi_get_capabilities (NM_DEVICE_WIFI (self->device));
@@ -1227,8 +1222,6 @@ net_device_wifi_new (CcPanel *panel, NMClient *client, NMDevice *device)
                                  G_CALLBACK (ap_activated), self, G_CONNECT_SWAPPED);
         g_signal_connect_object (list, "configure",
                                  G_CALLBACK (show_details_for_row), self, G_CONNECT_SWAPPED);
-        g_signal_connect_object (client, "notify",
-                                 G_CALLBACK(nm_client_on_permission_change), self, G_CONNECT_SWAPPED);
 
         nm_client_on_permission_change(self);
 
-- 
2.28.0

